Source/diablo.cpp

```diff
@@ -1142,43 +1141,42 @@ void CreateLevel(lvl_entry lvldir)
 	case DTYPE_TOWN:
 		CreateTown(lvldir);
 		InitTownTriggers();
-		LoadRndLvlPal(DTYPE_TOWN);
 		break;
 	case DTYPE_CATHEDRAL:
 		CreateL5Dungeon(glSeedTbl[currlevel], lvldir);
 		InitL1Triggers();
 		Freeupstairs();
-		if (currlevel < 21) {
-			LoadRndLvlPal(DTYPE_CATHEDRAL);
-		} else {
-			LoadRndLvlPal(DTYPE_CRYPT);
-		}
 		break;
 	case DTYPE_CATACOMBS:
 		CreateL2Dungeon(glSeedTbl[currlevel], lvldir);
 		InitL2Triggers();
 		Freeupstairs();
-		LoadRndLvlPal(DTYPE_CATACOMBS);
 		break;
 	case DTYPE_CAVES:
 		CreateL3Dungeon(glSeedTbl[currlevel], lvldir);
 		InitL3Triggers();
 		Freeupstairs();
-		if (currlevel < 17) {
-			LoadRndLvlPal(DTYPE_CAVES);
-		} else {
-			LoadRndLvlPal(DTYPE_NEST);
-		}
 		break;
 	case DTYPE_HELL:
 		CreateL4Dungeon(glSeedTbl[currlevel], lvldir);
 		InitL4Triggers();
 		Freeupstairs();
-		LoadRndLvlPal(DTYPE_HELL);
+		break;
+	case DTYPE_NEST:

```

Comment:
Not sure about the code style of this project as I haven't committed anything yet, but couldn't the cases DTYPE_CAVES and DTYPE_NEST as well as DTYPE_CATHEDRAL and DTYPE_CRYPT be merged as they're identical?
```
	case DTYPE_CATHEDRAL:
	case DTYPE_CRYPT:
		CreateL5Dungeon(glSeedTbl[currlevel], lvldir);
		InitL5Triggers();
		Freeupstairs();
		break;
	case DTYPE_CAVES:
	case DTYPE_NEST:
		CreateL3Dungeon(glSeedTbl[currlevel], lvldir);
		InitL3Triggers();
		Freeupstairs();
		break;
```

Reply:
I think it makes sense not to combine them, so the level generation can be separated and modified separately, if not now, then in the future.

Reply:
That's fine with me too, but shouldn't we split the cases in [Source/trigs.cpp](https://github.com/diasurgical/devilutionX/pull/1031/files#diff-4136c23fd95ecf1a6d9f9fafd598072a38ccc65de4ff352b550dc44758099cda) then too?

Reply:
Probably makes sense.

Reply:
Going to go with the combined condition :) It feels like the right choice for staying DRY.